fms/models/hf/utils.py

```diff
@@ -130,3 +134,93 @@ def to_hf_api(model: nn.Module, **override_config_kwargs) -> "HFModelArchitectur
+        the local path, otherwise the huggingface cache will be checked. If the huggingface cache does not contain the
+        model, then the weights will be downloaded and stored into the huggingface cache
+    device: Union[str, torch.device]
+        the device to load the model weights to
+
+    Returns
+    -------
+    nn.Module
+        an fms equivalent implementation of an HF model
+    """
+    # if the path does not exist, download it from huggingface and get the local path
+    if not os.path.exists(model_id_or_path):
+        model_path = snapshot_download(repo_id=model_id_or_path)
+    else:
+        model_path = model_id_or_path
+
+    config = AutoConfig.from_pretrained(model_path)
+
+    architecture = config.architectures[0]
+    config_params = {}
+
+    if architecture == "LlamaForCausalLM":
+        inner_dim = config.intermediate_size
+        architecture = "llama"
+        config_params["attn_bias"] = getattr(config, "attention_bias", False)
+        config_params["mlp_bias"] = getattr(config, "mlp_bias", False)
+        config_params["kv_heads"] = config.num_key_value_heads
+        config_params["norm_eps"] = config.rms_norm_eps
+        config_params["multiple_of"] = 1
+        config_params["emb_dim"] = config.hidden_size
+        config_params["max_expected_seq_len"] = config.max_position_embeddings
+    elif architecture == "GPTBigCodeForCausalLM":
+        inner_dim = config.n_inner
+        architecture = "gpt_bigcode"
+        config_params["ln_eps"] = config.layer_norm_epsilon
+        config_params["multiquery_attn"] = config.multi_query
+        config_params["emb_dim"] = config.hidden_size
+        config_params["max_expected_seq_len"] = config.n_positions
+    elif architecture == "MixtralForCausalLM":
+        inner_dim = config.intermediate_size
+        architecture = "mixtral"
+        config_params["dim"] = config.hidden_size
+        config_params["hidden_dim"] = inner_dim
+        config_params["norm_eps"] = config.rms_norm_eps
+        config_params["kv_heads"] = config.num_key_value_heads
+        config_params["num_experts"] = config.num_local_experts
+        config_params["top_k_experts"] = config.num_experts_per_tok
+        config_params["rope_base"] = config.rope_theta
+        config_params["max_expected_seq_len"] = config.max_position_embeddings
+    elif architecture == "RobertaForMaskedLM":
+        inner_dim = config.intermediate_size
+        architecture = "roberta"
+        config_params["emb_dim"] = config.hidden_size
+        config_params["pad_id"] = config.pad_token_id
+        config_params["max_pos"] = config.max_position_embeddings - 2
+        config_params["p_dropout"] = 0.0  # config.hidden_dropout_prob
+        config_params["norm_eps"] = config.layer_norm_eps
+        config_params["activation_fn"] = config.hidden_act
+    else:
+        raise ValueError(
+            "FMS model implementations currently only support LlamaForCausalLM and GPTBigCodeForCausalLM"
```

Comment:
we support more than Llama and GPTBigcode

Reply:
Fixed in latest commit